csharp-basic-topics/ActionAndFuncDelegates/ActionAndFuncDelegatesTests/ActionAndFuncTests.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IO;
+using CodeMaze;
+using Xunit;
+
+namespace ActionAndFuncDelegatesTests
```

Comment:
The tests are okay. In the future you should maybe take into account the main source code since we want to make sure everything works. When you write new examples for the tests, we still don't know if the source code from the main project is working or not. 
The alternative would be writing tests in such a way that they can be used as article code snippets. That's something we suggest for those that are good with writing tests.
Your choice, but we definitely need to be able to make sure the entire source code is working as expected without having to run each one ourselves.

Reply:
Hmm, your alternative idea makes sense. I will need to revise my article anyway, so this is the right time I improve the project as you suggested.

Reply:
Thanks, everything looks fine. I've put the idea out there although it's still your evaluation article. Your code is properly organized and formatted and that's what matters the most. You should use whatever you're most comfortable with.